desktop/core/src/desktop/models.py

```diff
@@ -1068,9 +1068,20 @@ def get_home_directory(self, user):
 
   def get_gist_directory(self, user):
     home_dir = self.get_home_directory(user)
-    gist_dir, created = Directory.objects.get_or_create(name=Document2.GIST_DIR, owner=user, parent_directory=home_dir)
-    if created:
-      LOG.info('Successfully created gist directory for user: %s' % user.username)
+
+    try:
+      gist_dir, created = Directory.objects.get_or_create(name=Document2.GIST_DIR, owner=user, parent_directory=home_dir)
+      if created:
+        LOG.info('Successfully created gist directory for user: %s' % user.username)
+    except Directory.MultipleObjectsReturned:
+      LOG.exception('Multiple Gist directories detected. Merging all into one.')
+
+      gist_dir = self.filter(owner=user, parent_directory=home_dir, name=Document2.GIST_DIR, type='directory').latest('-last_modified')
+      gist_dirs = self.filter(owner=user, parent_directory=home_dir, name=Document2.GIST_DIR, type='directory').exclude(uuid=gist_dir.uuid)
+      for dir in gist_dirs:
+        dir.children.exclude(name='.Trash').update(parent_directory=gist_dir)
+        dir.delete()
```

Comment:
For me this could still be one line of queryset (e.g DELETE ... id != last of first GistDirId)

Reply:
Confused and not able to get as to `how to update parent_dir of children of all duplicate gist dirs and then delete them too with one line of queryset.` Any inspiration or hints?? @romainr

Reply:
Does `one line of queryset` means hitting the database only once to update the parent_dir of all the children of dup gist directories and also deleting those dup gist directories? Because currently I am able to think of hitting the database twice for this task @romainr

Reply:
It should, but the main idea is to keep the code as concise as possible while staying simple (and not mix matching SQL + Python).

Reply:
We can keep it like this, the code is simple enough.

Was just curious if the update + delete operations could have been done combined into two querysets:

```
gist_dirs_dup.children_set.filter(Trash).update(...)
gist_dirs_dup.delete()
```